### PR TITLE
Docs updated with an example on JWKS and caching

### LINKED ISSUE #62: Add a cache for async keys
in `createVerified`, we accept a `key` as a function. However we do not cache the fetched key. Adding caching would likely speed things up quite a bit in case of remote jwks keys.

### LINKED ISSUE #63: Support for JWKS public keys
We should support fetching from a JWKS url and cache things.

https://github.com/nearform/fast-jwt/issues/62 for the caching bit.

### PR DESCRIPTION
**Pull request by sameer-coder**
Added an example on fetching from a JWKS and caching the key for better performance

Closes #62 
Closes #63

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -160,6 +160,30 @@ async function test() {
...
   // => { a: 1, b: 2, c: 3, iat: 1579521212 }
 }
+
+// Verify using JWKS with key as a function and caching enabled (Uses get-jwks package)
+const { createVerifier } = require('fast-jwt')
+const buildGetJwks = require('get-jwks')
+
+// well known url of the token issuer
+// often encoded as the `iss` property of the token payload
+const domain = 'https://...'
```
Comment (mcollina) [inline README.md:170]: please use the `allowedDomains` option of `buildGetJwks`.

File: README.md [hunk 2]
```diff
@@ -160,6 +160,30 @@ async function test() {
...
+
+const getJwks = buildGetJwks()
+
+// create a verifier function with key as a function
+const verifyWithPromise = createVerifier({
+  key: async function (token) {
+    const publicKey = getJwks.getPublicKey({
```
Comment (mcollina) [inline README.md:177]:
```suggestion
    const publicKey = await getJwks.getPublicKey({
```

File: README.md [hunk 3]
```diff
@@ -160,6 +160,30 @@ async function test() {
   const payload = await verifyWithPromise(token)
   // => { a: 1, b: 2, c: 3, iat: 1579521212 }
 }
+
```
Comment (mcollina) [inline README.md:163]: This should probably be a seprate code block.

**Review by mcollina — CHANGES_REQUESTED** [review]

**Review by simoneb — CHANGES_REQUESTED** [review]
I wouldn't duplicate the examples in both repos. We have the integration example in get-jwks, I would keep it there since we already mention in the docs of this repo that jwks integration can be done with that package. If we want to duplicate them, we should at least make sure they're in sync, but I would not duplicate them anyway.

File: README.md (hunk 1 above)
Comment (sameer-coder) [reply README.md:170]: I need the ```domain``` variable to pass it in ```getJwks.getPublicKey()``` when creating the verifier

Do you mean to also use ```allowedDomains``` option in ```buildGetJwks```, something like this?
```js
const getJwks = buildGetJwks({ allowedDomains: [domain]})
```

File: README.md (hunk 3 above)
Comment (sameer-coder) [reply README.md:163]: Yep! Agreed

**Comment by sameer-coder** [comment]
Yes @simoneb. I was thinking about that too. Do you want me to update the example on get-jwks instead of here and link that example in the docs here?

**Comment by simoneb** [comment]
> Yes @simoneb. I was thinking about that too. Do you want me to update the example on get-jwks instead of here and link that example in the docs here?

I would avoid duplication, yes.

**Pull request closed without merging**